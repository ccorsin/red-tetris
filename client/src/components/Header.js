import React from 'react';
import styled from 'styled-components';

const StyledHeader = styled.div`
    display: flex;
    flex-direction: column;
    margin-left: auto;
    margin-right: auto;
    width: 25%;
    filter: grayscale(0.95);
    margin-top: 10px;
    margin-bottom: 10px;
    font-family: Pixel, Arial, Helvetica, sans-serif;
`;

const Header = ({ playerCount, commands }) => {
    let count;
    if (playerCount > 1) {
        count = playerCount + " players in the game.";
    } else {
        count = playerCount + " player in the game.";
    }
      return (
        <StyledHeader>
            <img src ="https://www.pngkit.com/png/full/273-2736039_517-name-of-tetris-shapes-263-colorfulness.png" width="100%"/>
            <h1>{count}</h1>
            {commands}
        </StyledHeader>
    );
};

export default Header;
