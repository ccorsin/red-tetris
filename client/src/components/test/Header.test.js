import React from 'react'
import { shallow } from 'enzyme';
// import { Tetriminos } from '../../../server/models'
import Header from '../Header'
import toJson from "enzyme-to-json"

describe('<Header/> Component', () => {
  it('rendering correctly with no shapes', () => {
    let wrapper = shallow(
      <Header playerCount='1' commands='0'/>
    )
    expect(wrapper).toMatchSnapshot();
  });
  // it('rendering correctly with shapes', () => {
  //   // const tetriminos = new Tetriminos
  //   // let tetro = []
  //   // for (let i = 0; i < 5; i++) {
  //   //   tetro.push(tetriminos.randomTetromino())
  //   // }
  //   let wrapper = shallow(
  //     <Header playerCount='1' commands='0'/>
  //   )
  //   expect(toJson(wrapper)).toMatchSnapshot()
  // })
})