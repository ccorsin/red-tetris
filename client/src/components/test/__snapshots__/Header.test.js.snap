// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Component rendering correctly with no shapes 1`] = `ShallowWrapper {}`;
